protoc_plugin/test/to_builder_test.dart

```diff
@@ -96,4 +104,83 @@ main() {
+    EmptyMessage emptyMessage;
+    int tagNumber;
+    UnknownFieldSet unknownFieldSet;
+    UnknownFieldSetField field;
+
+    setUp(() {
+      inner = Inner()..value = 'bob';
+      emptyMessage = EmptyMessage.fromBuffer(inner.writeToBuffer());
+      tagNumber = inner.getTagNumber('value');
+      unknownFieldSet = emptyMessage.unknownFields;
+      field = unknownFieldSet.getField(tagNumber);
+    });
+
+    test('can read from a frozen unknown fieldset', () {
+      expect(unknownFieldSet.hasField(tagNumber), isTrue);
+      expect(field.lengthDelimited[0], utf8.encode(inner.value));
+
+      emptyMessage.freeze();
+      unknownFieldSet = emptyMessage.unknownFields;
+      field = unknownFieldSet.getField(tagNumber);
+
+      expect(unknownFieldSet.hasField(tagNumber), isTrue);
+      expect(field.lengthDelimited[0], utf8.encode(inner.value));
+    });
+
+    test('cannot add to a frozen UnknownFieldSetField', () {
+      emptyMessage.freeze();
+
+      expect(
+          () => field.addFixed32(1), throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.fixed32s.add(1),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.addFixed64(Int64(1)),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.fixed64s.add(Int64(1)),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.addLengthDelimited([1]),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.lengthDelimited.add([1]),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.addGroup(unknownFieldSet.clone()),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.groups.add(unknownFieldSet.clone()),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.addVarint(Int64(1)),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => field.varints.add(Int64(1)),
+          throwsA(TypeMatcher<UnsupportedError>()));
+    });
+
+    test('cannot add or merge field to a frozen UnknownFieldSet', () {
+      emptyMessage.freeze();
+
+      expect(() => unknownFieldSet.addField(2, field),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => unknownFieldSet.mergeField(2, field),
+          throwsA(TypeMatcher<UnsupportedError>()));
+    });
+
+    test('cannot merge message into a frozen UnknownFieldSet', () {
+      emptyMessage.freeze();
+      EmptyMessage other = emptyMessage.clone();
+
+      expect(() => emptyMessage.mergeFromBuffer(other.writeToBuffer()),
+          throwsA(TypeMatcher<UnsupportedError>()));
+      expect(() => emptyMessage.mergeFromMessage(other),
+          throwsA(TypeMatcher<UnsupportedError>()));
+    });
+
+    test('cannot add a field to a frozen UnknownFieldSet', () {
+      emptyMessage.freeze();
+
+      expect(() => unknownFieldSet.addField(tagNumber, field),
+          throwsA(TypeMatcher<UnsupportedError>()));
+    });
```

Comment:
Could you add tests that toBuilder correctly copies an UnknownFieldSet to a mutable one?
This could be here or in `to_builder_test.dart`